build.gradle

```diff
@@ -142,19 +145,55 @@ tasks.withType(JavaCompile).configureEach {
 	options.compilerArgs.add '-XDstringConcat=inline'
 }
 
-build.dependsOn('installLocalGitHookPostCheckout')
-tasks.register('installLocalGitHookPostCheckout', Copy) {
-	group = 'git'
-	from new File(rootProject.rootDir, '.scripts/post-checkout')
-	into { new File(rootProject.rootDir, '.git/hooks') }
-	filePermissions {
-		unix(0775)
+
+// Setup local git hooks
+
+def ensureSymlink(link, target) {
+	if (!Files.isSymbolicLink(link)) {
+		Files.deleteIfExists(link)
+
+		Files.createSymbolicLink(link, target)
+
+		def relative = { f -> rootDir.toPath().relativize(f) }
+		println("--- Installed symbolic link ${relative(link)} -> ${relative(target)} ---")
 	}
+}
+
+def gitConfig = 'Git Configuration'
+tasks.register('installLocalGitHookSymlinks') {
+	group = gitConfig
+
+	inputs.dir "$rootDir/.scripts/git-hooks/"
+	inputs.dir "$rootDir/.git/hooks/"
+	outputs.dir "$rootDir/.git/hooks/"
+
 	doLast {
-		println("---Added Post Checkout Git Hook Script---")
+		def hooksSources = fileTree("$rootDir/.scripts/git-hooks/")
+		def hooksDest = file("$rootDir/.git/hooks/").toPath()
+
+		hooksSources.forEach { File target ->
+			def link = hooksDest.resolve(target.toPath().getFileName())
+
+			ensureSymlink(link, target.toPath())
+		}
+
+		hooksDest.toFile().listFiles().each { link ->
+			if (Files.isSymbolicLink(link.toPath()) && !Files.exists(Files.readSymbolicLink(link.toPath()))) {
+				link.delete()
+				println("--- Deleted stale link ${rootDir.toPath().relativize(link.toPath())} ---")
+			}
+		}
 	}
 }
 
+def gitConfigTasks = tasks.matching { it.group == gitConfig }
+tasks.matching({ it.group != gitConfig }).configureEach({ it.dependsOn gitConfigTasks })
+
+if (System.getenv().containsKey('CI') || !file('.git').isDirectory()) {
```

Comment:
when the if itself run, on build, on refresh gradle?

Reply:
I believe on gradle refresh